assembly/src/parser/lexer.rs

```diff
@@ -609,8 +636,31 @@ fn parse_hex(span: SourceSpan, hex_digits: &str) -> Result<HexEncodedValue, Pars
     }
 }
 
+fn parse_bin(span: SourceSpan, bin_digits: &str) -> Result<BinEncodedValue, ParsingError> {
+    if bin_digits.len() <= 32 {
+        let value =
+            u32::from_str_radix(bin_digits, 2).map_err(|error| ParsingError::InvalidLiteral {
+                span,
+                kind: int_error_kind_to_literal_error_kind(
+                    error.kind(),
+                    LiteralErrorKind::FeltOverflow,
+                ),
+            })?;
+        Ok(shrink_u32_bin(value))
+    } else {
+        Err(ParsingError::InvalidBinaryLiteral {
+            span,
+            kind: BinErrorKind::TooLong,
+        })
+    }
+}
+
+fn is_ascii_binary(c: char) -> bool {
```

Comment:
**Suggestion:** Mark this `#[inline(always)]` to ensure that it gets inlined, otherwise it goes from a very cheap comparison operation to a quite expensive function call if it isn't inlined. While Rust may inline functions like this automatically, it is based on a heuristic, and is sensitive to what is happening in the caller, so in cases where you know it is important, it's always worth providing the inline annotation